File: Data Structures & Algorithms/duplicate-integer/submission-5.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_set <int> s;
        if (nums.size()<2)
        {
            return false;
        }
        else 
        for(int num : nums)
        {
            if (s.count(num))
            {
                return true;
            }
            s.insert(num);
        }
        return false;
    }};

File: Data Structures & Algorithms/valid-palindrome-ii/submission-5.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        l,r = 0,len(s)-1
        if s == s[::-1]:
            return True
        else:
            while(l < r):
                if (s[l] == s[r]):
                    l+=1
                    r-=1
                else:
                    sl = s[l+1:r+1]
                    sr = s[l:r]
                    return sl == sl[::-1] or sr[::-1] == sr
        return True

        